nova/scheduler/weights/__init__.py

```diff
@@ -35,13 +35,24 @@ class BaseHostWeigher(weights.BaseWeigher):
     """Base class for host weights."""
     pass
 
+    def host_info_for_instance_ids(self, _):
+        return set()
+
 
 class HostWeightHandler(weights.BaseWeightHandler):
     object_class = WeighedHost
 
     def __init__(self):
         super(HostWeightHandler, self).__init__(BaseHostWeigher)
 
+    @staticmethod
+    def host_info_for_instance_ids(filters, request_spec):
+        instance_ids = set()
+        for filter_ in filters:
+            instance_ids.update(
+                filter_.host_info_for_instance_ids(request_spec))
+        return instance_ids
```

Comment:
Should be `weighers` instead of `filters`.

Reply:
Done